src/btree/bt_discard.c

```diff
@@ -243,6 +243,12 @@ __wt_free_ref(WT_SESSION_IMPL *session, WT_REF *ref, int page_type, bool free_pa
     if (ref == NULL)
         return;
 
+    /*
+     * Flagging the page type in the WT_REF is a change, and the only current user of WT_REF.flags.
+     * Assert for now, to ensure we find all paths which allocate a WT_REF (02/2020).
```

Comment:
Could you reword this to be more permanent? I don't see why we wouldn't always want this. Maybe `Ensure WT_REF structure is properly initialized - there are many places that create WT_REFs, and they must always refer to either a leaf or internal page.`

Reply:
Done, and the test reworked to match.

I don't see any reason to remove this unless we run out of bits  in `WT_REF.flags`, in which case we could lose one of these bits, only one is necessary.

Pushed in 9f1815a8e.